docs/configuration.md

```diff
@@ -0,0 +1,40 @@
+# Configuration
+
+## Configuration files
+
+ty supports persistent configuration files at both the project- and user-level.
+
+Specifically, ty will search for a `pyproject.toml` or `ty.toml` file in the current directory, or in the nearest parent directory.
+
+If a `pyproject.toml` file is found, ty will read configuration from the `[tool.ty]` table. For example, to ignore the `index-out-of-bounds` rule, add the following to a `pyproject.toml`:
+
+**`pyproject.toml`**:
+
+```toml
+[tool.ty.rules]
+index-out-of-bounds = "ignore"
+```
+
+(If there is no `tool.ty` table, the `pyproject.toml` file will be ignored, and ty will continue searching in the directory hierarchy.)
+
+ty will also search for `ty.toml` files, which follow an identical structure, but omit the `[tool.ty]` prefix. For example:
+
+**`ty.toml`**:
+
+```toml
+[rules]
+index-out-of-bounds = "ignore"
+```
+
+> [!NOTE]
+> `ty.toml` files take precedence over `pyproject.toml` files, so if both `ty.toml` and `pyproject.toml` files are present in a directory, configuration will be read from `ty.toml`, and the `[tool.ty]` section in the accompanying `pyproject.toml` will be ignored.
```

Comment:
We'll need to rewrite these to use `!!! note` admonitions instead. This is a follow-up task noted in #744